validator_worker/src/sentry_interface.rs

```diff
@@ -33,22 +34,18 @@ impl<T: Adapter + 'static> SentryApi<T> {
         channel: &Channel,
         config: &Config,
         logging: bool,
-        whoami: &str,
+        whoami: &ValidatorId,
     ) -> Result<Self, ValidatorWorker> {
         let client = Client::builder()
             .timeout(Duration::from_secs(config.fetch_timeout.into()))
             .build()
             .unwrap();
 
         // validate that we are to validate the channel
-        match channel
-            .spec
-            .validators
-            .into_iter()
-            .find(|&v| v.id == whoami)
-        {
-            Some(v) => {
-                let validator_url = format!("{}/channel/{}", v.url, channel.id);
+        match channel.spec.validators.find(whoami.clone()) {
+            SpecValidator::Leader(v) | SpecValidator::Follower(v) => {
+                let channel_id = format!("0x{}", hex::encode(&channel.id));
```

Comment:
Again the trick with SerHex can be used :stuck_out_tongue: and one more down bellow.

Reply:
The SerHex crate isn't used in validator_worker

Reply:
ok, cool!